File: swift/Sources/ShapesClientShell/InputReader.swift
import Foundation

class InputReader {
    public static func readInput() -> String {
        print("Your Answer >>> ", terminator: "")
        var userInput = readLine()
        userInput = userInput?.trimmingCharacters(in: .whitespaces).lowercased()
        readExit(userInput!)
        return userInput!
    }

    public static func readNumber() -> Double {
        while true {
            let userInput = readInput()

            if let number = Double(userInput) {
                return number
            } else {
                print("Invalid input. Please enter a valid number.")
            }
        }
    }

    private static func readExit(_ userInput: String) {
        if userInput == "exit" || userInput == "e" {
            print("Thank you for using the Shapes CLI!")
            exit(0)
        }
    }
}

File: csharp/src-temp/Shapes/Client/Shell/InputReader.cs
namespace Shapes.Client.Shell;

sealed class InputReader {
    public static string ReadInput() {
        Console.Write("Your Answer >>> ");
        string? userInput = Console.ReadLine()!;
        userInput = userInput.Trim().ToLower();
        ReadExit(userInput);
        return userInput;
    }

    public static double ReadNumber() {
        while (true) {
            string userInput = ReadInput();

            if (double.TryParse(userInput, out double result)) {
                return result;
            } else {
                Console.WriteLine("Invalid number. Please try again.");
            }
        }
    }

    private static void ReadExit(string userInput) {
        if (userInput.Equals("exit") || userInput.Equals("e")) {
            Console.WriteLine("Thank you for using the Shapes CLI!");
            Environment.Exit(0);
        }
    }
}
